### PR TITLE
feature/PP-171 

### PR DESCRIPTION
**Pull request by franciscotobar**
## What

- Created a transaction history modal to be able to verify the transactions for each smart wallet

## Why

- With the current implementation, each time the users try to execute a transaction, they need to wait until it is confirmed (or rejected). To avoid this, we could provide the user with the link to the explorer.

## Refs

- [Jira issue](https://rsklabs.atlassian.net/browse/PP-171)
- https://github.com/rsksmart/rif-relay-client/pull/20
- https://github.com/rsksmart/rif-relay-sdk/pull/25
- https://github.com/rsksmart/rif-relay-common/pull/23

### CODE REVIEW

File: src/context/reducer.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { Action, State } from 'src/context/types';
+
+const StoreReducer = (state: State, action: Action) => {
+    switch (action.type) {
+        case 'set_account':
```
Comment (antomor) [inline src/context/reducer.tsx]: I think that we may store all the actions in constants to be reused elsewhere

File: src/Utils.ts [hunk 2]
```diff
@@ -98,6 +99,22 @@ class Utils {
...
+        window.open(
+            `${process.env.REACT_APP_BLOCK_EXPLORER}/tx/${trx}`,
+            '_blank'
+        );
+    }
+
+    static addTransaction(address: string, transaction: Transaction) {
+        let transactions: Transaction[] = [];
+        if (address in localStorage) {
+            transactions = JSON.parse(localStorage.getItem(address)!);
```
Comment (antomor) [inline src/Utils.ts]: I'd use a try-catch here, just in case we read some dirty data from the localstorage, we may need to reset it.

**Review by antomor — CHANGES_REQUESTED** [review]

File: src/modals/Transfer.tsx [hunk 3]
```diff
@@ -128,7 +128,8 @@ function Transfer(props: TransferProps) {
             };
 
             const txDetails = await state.provider!.relayTransaction(
-                relayTrxOpts
+                relayTrxOpts,
+                true
```
Comment (antomor) [inline src/modals/Transfer.tsx]: In order to use this parameter in this way, we may need to change the SDK [interface](https://github.com/rsksmart/rif-relay-sdk/blob/master/src/index.ts#L70) and [method](https://github.com/rsksmart/rif-relay-sdk/blob/master/src/sdk.ts#L302). Maybe we could consider just adding an optional parameter in the [RelayTransactionOptions](https://github.com/rsksmart/rif-relay-sdk/blob/master/src/interfaces.ts#L52).
Furthermore, have you considered supporting the exact mechanism for the deployment?

**Review by antomor — CHANGES_REQUESTED** [review]
@franciscotobar thank you for the changes.
Apart from the comments I left, maybe we should avoid waiting for the transaction receipt by default, so we don't need to change all the `relayTransaction` calls in the dapp. WDYT?

File: src/modals/Transfer.tsx (hunk 3 above)
Comment (franciscotobar) [reply src/modals/Transfer.tsx]: In the deployment, the logic for waiting for the receipt is on the dapp. When you refer to this, you mean to move the logic to the rif-relay-client so they can have the option to decide to wait but from the API itself

File: src/context/reducer.tsx (hunk 1 above)
Comment (franciscotobar) [reply src/context/reducer.tsx]: there is a types.ts class that stores all the actions and constants, inside the context folder.

File: src/Utils.ts (hunk 2 above)
Comment (franciscotobar) [reply src/Utils.ts]: I agree with you, let me add the try/catch

**Comment by antomor** [comment]
Although LGTM, I wasn't able to run it, because of the following error: `FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory`. I don't know if we just introduced the issue, because when I tried some days ago, I didn't have any issue

**Comment by antomor** [comment]
> Although LGTM, I wasn't able to run it, because of the following error: `FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory`. I don't know if we just introduced the issue, because when I tried some days ago, I didn't have any issue

I managed to run it using node v14. LGTM!

File: src/context/reducer.tsx (hunk 1 above)
Comment (antomor) [reply src/context/reducer.tsx]: I mean, couldn't we store the strings into constant variables? something like
```
const SET_ACCOUNT_ACTION = 'set_account';
const SET_CONNECTED_ACTION = 'set_connected';
...
```
WDYT?

File: src/modals/Transfer.tsx (hunk 3 above)
Comment (antomor) [reply src/modals/Transfer.tsx]: After discussing, we managed to include the same logic for the deployment as well. Thank you for the changes!

File: src/context/reducer.tsx (hunk 1 above)
Comment (franciscotobar) [reply src/context/reducer.tsx]: sure, we could do that, let me do the changes and push it.

Comment (franciscotobar) [reply src/context/reducer.tsx]: I was looking for alternatives and we could use an enum too, let me know what do you think about it.

Comment (antomor) [reply src/context/reducer.tsx]: I find the solutions are almost the same. Let's keep the current implementation.

**Review by antomor — APPROVED** [review]
LGTM! @franciscotobar Thank you for these changes!

**Pull request merged**